### ISSUE TITLE
Solidity functions which take an array parameter get an error at runtime

### ISSUE THREAD

**Issue by sdesmond46** [Neufund/TypeChain#39]
It looks like in the generation code which creates a new `DeferredTransactionWrapper`, all input arguments have their `toString` method called on them. This works fine normally, but for array arguments the web3 formatting runtime gives an error trying to iterate over a string when it expects and array.

Given this Contract:

```solidity
contract TypeChainBugDemo {

  uint256 public _val;

  // Calling this function from the generated TypeChain bindings will fail
  function callWithArray(uint256[] arrayArg) public {
    _val = arrayArg[0];
  }
}
```

The `callWithArray` function will generate the following TS definition:
```typescript
  public callWithArrayTx(
    arrayArg: BigNumber[]
  ): DeferredTransactionWrapper<ITxParams> {
    return new DeferredTransactionWrapper<ITxParams>(this, "callWithArray", [
      arrayArg.toString()
    ]);
  }
```

The line `arrayArg.toString()` will convert our array into a single string. web3 eventually throws an error in `SolidityType.prototype.encode` when it attempts to operate on that string thinking it's an array

```
TypeError: value.forEach is not a function
    at /Users/sdesmond/GitHub/EVNT/truffle/node_modules/web3/lib/solidity/type.js:165:19
    at SolidityTypeUInt.SolidityType.encode (/Users/sdesmond/GitHub/EVNT/truffle/node_modules/web3/lib/solidity/type.js:170:11)
    at /Users/sdesmond/GitHub/EVNT/truffle/node_modules/web3/lib/solidity/coder.js:91:29
    at Array.map (<anonymous>)
    at SolidityCoder.encodeParams (/Users/sdesmond/GitHub/EVNT/truffle/node_modules/web3/lib/solidity/coder.js:90:34)
    at SolidityFunction.toPayload (/Users/sdesmond/GitHub/EVNT/truffle/node_modules/web3/lib/web3/function.js:92:52)
    at SolidityFunction.sendTransaction (/Users/sdesmond/GitHub/EVNT/truffle/node_modules/web3/lib/web3/function.js:163:24)
    at /Users/sdesmond/GitHub/EVNT/truffle/gen-src/typechain-runtime.ts:58:9
```

**Comment by krzkaczor** [comment]
Fixed by https://github.com/Neufund/TypeChain/pull/40. Thanks for contribution!

**Comment by krzkaczor** [comment]
Released as `0.1.5`
